### ISSUE TITLE
Previous firmware versions.

### ISSUE THREAD

**Issue by AlexGhiti** [lgrangeia/tomtom-hacking#2]
Hi, 

Sorry to have to open an issue whereas it is not really one. I read very carefully your posts on the tomtom runner hacking and it was very amazing, thanks for that. But as you said in your last post, tomtom solved the issues you took advantage of in the firmware 1.8.46. I did not find any way to get older firmwares, do you have any idea how to get them ?

Sorry again to bother you, I did not find any other way to get to you :)

Alex

**Comment by lgrangeia** [comment]
Hi Alex,

I'd try something like this. Let me know what are your results :)

If you want to get my attention faster mention or DM me on twitter.

Regards,

LG

``` python
#!/usr/bin/python
import os

maj = 1
mid = 0
min = 0

while (maj <= 1):
  while (mid <= 10):
    while (min <= 60):
      cmd = 'wget http://download-test.tomtom.com/sweet/fitness/Firmware/E9030000/{:d}_{:d}_{:d}'.format(maj,mid,min) + '/0x000000F0 -O {:d}.{:d}.{:d}-0x000000F0'.format(maj,mid,min)
      os.system(cmd)
      min += 1
    min = 0
    mid += 1
  mid = 0
  maj += 1

```

**Comment by AlexGhiti** [comment]
Oh thanks, I had tried that manually but on download.tomtom.com and not download-test.tomtom.com.

Many thanks, especially for your posts, I learned a lot :)

**Comment by arvinddoraiswamy** [comment]
Yes, thank you indeed. I did exactly what @AlexGhiti did above :)

I was wondering about the same thing and searching for an Email address to mail you at. I could modify the numbers in the script a bit and download older firmware.

A great series of posts and I am learning so much from them. Thank you very much. Only thing left is to get a second-hand TomTom watch and actually try this out now :)
